pandas/_libs/algos.pyx

```diff
@@ -998,12 +998,13 @@ def rank_1d(
         TiebreakEnumType tiebreak
         Py_ssize_t i, j, N, grp_start=0, dups=0, sum_ranks=0
         Py_ssize_t grp_vals_seen=1, grp_na_count=0
-        ndarray[int64_t, ndim=1] lexsort_indexer
-        ndarray[float64_t, ndim=1] grp_sizes, out
+        ndarray[int64_t, ndim=1] lexsort_indexer, grp_sizes
```

Comment:
lexsort_indexer[i] is being passed to `ndarray.__getitem__` -> it shouldbe intp_t

Reply:
good catch thanks